src/power_grid_model/data_types.py

```diff
@@ -7,84 +7,210 @@
+
+- data: a :class:`SingleArray`. The exact dtype depends on the type of component.
+- indptr: an :class:`IndexPointer` representing the start and end indices for each batch scenario.
 
 - Examples:
 
-    - structure: {"indptr": <1d-array>, "data": :class:`SingleArray`}
+    - structure: {"indptr": :class:`IndexPointer`, "data": :class:`SingleArray`}
     - concrete example: {"indptr": [0, 2, 2, 3], "data": [(0, 1, 1), (1, 1, 1), (0, 0, 0)]}
 
-        - the scenario 0 sets the statuses of components ids 0 and 1 to 1 (and keeps defaults for other components)
+        - the scenario 0 sets the statuses of components with ids 0 and 1 to 1 (and keeps defaults for other components)
         - scenario 1 keeps the default values for all components
         - scenario 2 sets the statuses of component with id 0 to 0 (and keeps defaults for other components)
 """
 
+SparseBatchColumnarData = dict[str, IndexPointer | SingleColumnarData]
+"""
+Sparse batch columnar data is a dictionary containing the keys `indptr` and `data`.
+
+- data: a :class:`SingleColumnarData`. The exact supported attribute columns depend on the component type.
+- indptr: an :class:`IndexPointer` representing the start and end indices for each batch scenario.
+
+- Examples:
+
+    - structure: {"indptr": :class:`IndexPointer`, "data": :class:`SingleArray`}
+    - concrete example: {"indptr": [0, 2, 2, 3], "data": {"id": [0, 1, 0], "status": [1, 1, 0]}}
+
+        - the scenario 0 sets the status of components with ids 0 and 1 to 1 (and keeps defaults for other components)
+        - scenario 1 keeps the default values for all components
+        - scenario 2 sets the status of component with id 0 to 0 (and keeps defaults for other components)
+"""
+
+DenseBatchData = DenseBatchArray | DenseBatchColumnarData
+"""
+Dense batch data can be a :class:`DenseBatchArray` or a :class:`DenseBatchColumnarData`.
+"""
+
+SparseBatchData = SparseBatchArray | SparseBatchColumnarData
+"""
+Sparse batch data can be a :class:`SparseBatchArray` or a :class:`SparseBatchColumnarData`.
+"""
+
 BatchArray = DenseBatchArray | SparseBatchArray
 """
 A batch array is a either a :class:`DenseBatchArray` or a :class:`SparseBatchArray`.
 """
 
+BatchColumnarData = DenseBatchColumnarData | SparseBatchColumnarData
+"""
+Batch columnar data is either a :class:`DenseBatchColumnarData` or a :class:`SparseBatchColumnarData`.
+"""
+
 DataArray = SingleArray | BatchArray
 """
 A data array can be a :class:`SingleArray` or a :class:`BatchArray`.
 """
 
-SingleDataset = dict[ComponentTypeVar, SingleArray]
+ColumnarData = SingleColumnarData | BatchColumnarData
+"""
+Columnar data can be :class:`SingleColumnarData` or :class:`BatchColumnarData`.
+"""
+
+# SingleComponentData = SingleArray | SingleColumnarData
+# """
+# Single component data can be :class:`SingleArray` or :class:`SingleColumnarData`.
+# """
+SingleComponentData = Union[SingleArray]
```

Comment:
Here and bellow: The commented lines should replace the non-commented once the columnar data stuff is ready on the Python API side, right? But, can't and shouldn't we implement/add the commented lines at this point?

Reply:
Never mind this comment, I just re-read your initial comment at the very top. Will leave here as a reminder, but feel free to resolve.